### DISCUSSION TITLE
How to use tailwind css or bootstrap in pages ?

### DISCUSSION THREAD

**Discussion by aeq-dev** [filamentphp/filament#3840] [Help] [answered]
Hello everyone,
I'm new in filament, I've created resources for models, and I need some pages for custom features, how can I use tailwind or bootstrap to set my components ?
I've followed this [https://filamentphp.com/docs/2.x/admin/appearance#building-themes](https://filamentphp.com/docs/2.x/admin/appearance#building-themes) but doesn't work, I'm on Laravel 9, vite
Thanks

**Comment by danharrin** [comment]
wdym it doesnt work?

**Comment by aeq-dev** [comment]
Thanks for reply : 

Here what I did : 
1. `npm install tailwindcss @tailwindcss/forms @tailwindcss/typography autoprefixer tippy.js --save-dev`
2. Create filament.css in `resources/css/filament.css` contains : 
`@import '../../vendor/filament/filament/resources/css/app.css';`

3. Vite : 
```
laravel({
            input: ['resources/css/app.css', 'resources/js/app.js','resources/css/filament.css'],
            
            refresh: true,
        }),
```
4. npm run build & dev
5. Create a page using `php artisan make:filament-page MyPage`
6. Edit my-page blade : 
```
<x-filament::page>
    <div class="p-4 mb-4 text-sm text-blue-700 bg-blue-100 rounded-lg dark:bg-blue-200 dark:text-blue-800" role="alert">
        <span class="font-medium">Info alert!</span> Change a few things up and try submitting again.
    </div>
</x-filament::page>
```
I doesn't render as expected

Thanks

**Reply by danharrin** [reply]
What is in your tailwind.config?

**Comment by aeq-dev** [comment]
Sorry for delay,
```
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: [
    "./resources/**/*.blade.php",
    "./resources/**/*.js",
    './vendor/filament/**/*.blade.php', 
  ],
  darkMode: 'class',
  theme: {
    extend: {},
  },
  plugins: [
    require('@tailwindcss/forms'), 
    require('@tailwindcss/typography'), 
  ],
}

```
Note that when I did `npm run build ` I got : 
```
[vite:css] /home/../vendor/filament/forms/dist/module.esm.css:1657:3: The `border-primary-500` class does not exist. If `border-primary-500` is a custom class, make sure it is defined within a `@layer` directive.
file: /home/.../resources/css/filament.css
```

**Reply by danharrin** [reply]
You need to add the primary, danger, warning, success colors, so that there isnt an error, otherwise your CSS wont build

**Reply by aeq-dev** [reply]
Yes, Thanks :)

**Comment by aeq-dev** [comment, marked as answer]
Never mind, I fixed it by editing tailwindcss config : 
```
const colors = require('tailwindcss/colors')
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: [
    "./resources/**/*.blade.php",
    //"./resources/**/*.js",
    './vendor/filament/**/*.blade.php', 
  ],
  darkMode: 'class',
  theme: {
    extend: {
      colors: {
        danger: colors.rose,
        primary: colors.blue,
        success: colors.green,
        warning: colors.yellow,
      },
    },
  },
  plugins: [
    require('@tailwindcss/forms'), 
    require('@tailwindcss/typography'), 
  ],
}

```
